### PR TITLE
tests: make optional the tf-trace / fix govcloud to get ips and other refactors

### PR DESCRIPTION
**Pull request by cpanato**

### CODE REVIEW

**Comment by enxebre** [comment]
lgtm

**Comment by squat** [comment]
ok to test

File: tests/rspec/lib/cluster.rb [hunk 1]
```diff
@@ -209,13 +208,28 @@ def terraform_init
...
+    Open3.popen3(env, "bash -co pipefail 'make -C ../.. #{cmd} |
+                    tee ../../build/#{@name}/terraform-#{cmd}.log'") do |_stdin, stdout, stderr, wait_thr|
+      while (line = stdout.gets)
+        puts line
+      end
+      while (line = stderr.gets)
+        puts line
+      end
+      exit_status = wait_thr.value
+      return true if exit_status.success?
```
Comment (mxinden) [inline tests/rspec/lib/cluster.rb]: `s/true if exit_status.success?/exit_status.success?/`

Comment (cpanato) [reply tests/rspec/lib/cluster.rb]: done

Comment (mxinden) [reply tests/rspec/lib/cluster.rb]: Let me know once you pushed the changes.

**Comment by cpanato** [comment]
ok to test

**Comment by cpanato** [comment]
retest this please. govcloud only

File: tests/rspec/lib/cluster.rb [hunk 2]
```diff
@@ -172,13 +172,12 @@ def license_and_pull_secret_defined?
...
-      3.times do |idx|
+      3.times do
         env = env_variables
-        env['TF_LOG'] = 'TRACE' if idx.positive?
         env['TF_APPLY_OPTIONS'] = '-no-color'
         env['TF_INIT_OPTIONS'] = '-no-color'
-        return true if system(env, "bash -co pipefail 'make -C ../.. apply |
-                                    tee ../../build/#{@name}/terraform-apply.log'")
+
+        return true if run_command(env, 'apply')
```
Comment (mxinden) [inline tests/rspec/lib/cluster.rb]: same here, how about just `return run_command(env, 'apply')`?

Comment (cpanato) [reply tests/rspec/lib/cluster.rb]: roger that

Comment (cpanato) [reply tests/rspec/lib/cluster.rb]: done @mxinden

**Pull request merged**